File: database/logic/usage.py
from database.index import DatabasePool

class UsageTable:

    def __init__(self, pool: DatabasePool):

        self.pool = pool

    def increaseCommandUsage(self, command: str) -> None:
        
        with self.pool.connection() as cursor:
            cursor.execute(
                """
                insert into usage (command, uses) values (%s, 1)
                on conflict(command) do update set uses = usage.uses + 1
                """, (command,)
            )
       
    
    def fetchCommands(self) -> list:
        
        with self.pool.connection() as cursor:
            
            cursor.execute("select * from usage order by uses desc")
            return cursor.fetchall()

File: src/database/logic/usage.ts
import { withDb } from "../pool";

type CommandUsage = {
  command: string;
  uses: number;
};

export class UsageTable {


  public async increaseCommandUsage(command: string): Promise<void> {

    await withDb(async (client) => {
      await client.query(
        `
        INSERT INTO usage (command, uses)
        VALUES ($1, 1)
        ON CONFLICT (command)
        DO UPDATE SET uses = usage.uses + 1
        `,
        [command]
      );
    });
  }

  public async fetchCommands(): Promise<CommandUsage[]> {

    return await withDb(async (client) => {
      const result = await client.query<CommandUsage>(
        `
        SELECT command, uses
        FROM usage
        ORDER BY uses DESC
        `
      );

      return result.rows;
    });
  }
}

export const usageTable = new UsageTable();